src/System.Threading/tests/SemaphoreTests.cs

```diff
@@ -74,15 +73,6 @@ public void Ctor_InvalidArguments()
             AssertExtensions.Throws<ArgumentException>(null, () => new Semaphore(2, 1, "CtorSemaphoreTest", out createdNew));
         }
 
-        [PlatformSpecific(TestPlatforms.Windows)]  // named semaphores aren't supported on Unix
-        [Fact]
-        public void Ctor_InvalidNames()
```

Comment:
I would guess that there is still limit on the name in the OS. Could you please try what happens for 10MB name?

Reply:
I tried with ```10,000,000 * 2 bytes = 20 megabytes``` and the test is still not failing

Reply:
Ok, you are right. I have tried as well and there does not seem to be any hard-coded limit.